pennylane/optimize/rotosolve.py

```diff
@@ -12,172 +12,452 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Rotosolve gradient free optimizer"""
+# pylint: disable=too-many-branches,cell-var-from-loop
 
 import numpy as np
+from scipy.optimize import brute, shgo
 from pennylane.utils import _flatten, unflatten
 
 
+def _assert_integer(x):
+    x_type = type(x)
+    if not np.issubdtype(x_type, np.integer):
+        raise ValueError(
+            "The numbers of frequencies are expected to be integers. " f"Received {x_type}."
+        )
+
+
+def _validate_num_freqs(num_freqs, requires_grad):
```

Comment:
what is `requires_grad` here? Is it a list of integers?

Reply:
It is a list of `bool`eans. I'll add a docstring!

Reply:
I see from further down that it is a list of bools :)